### PR TITLE
Add template HPA

### PR DESCRIPTION
**Pull request by mathieu-benoit**

### CODE REVIEW

**Comment by mathieu-benoit** [comment]
Ready for your review, thanks!

File: resource-definitions/template-driver/horizontalpodautoscaler/hpa-template.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+entity:
+  driver_type: humanitec/template
+  name: hpa-template
+  type: horizontal-pod-autoscaler
+  driver_inputs:
+    values:
+      templates:
+        init: |
+          workload: {{ index (splitList "." "${context.res.id}") 1 }}
+          maxReplicas: {{ .resource.maxReplicas | default 3 }}
```
Comment (astromechza) [inline resource-definitions/template-driver/horizontalpodautoscaler/hpa-template.yaml:14]: As an example, you could add the min() clamp here as well?

```suggestion
          maxReplicas: {{ .resource.maxReplicas | default 3 | min 10 }}

```

File: resource-definitions/template-driver/horizontalpodautoscaler/hpa-template.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  id: hpa-template
+entity:
+  driver_type: humanitec/template
+  name: hpa-template
+  type: horizontal-pod-autoscaler
+  driver_inputs:
+    values:
+      templates:
+        init: |
+          workload: {{ index (splitList "." "${context.res.id}") 1 }}
```
Comment (astromechza) [inline resource-definitions/template-driver/horizontalpodautoscaler/hpa-template.yaml:13]: For easier reading you could define the "constants" up top here
```suggestion
          {{ $defaultMaxReplicas := 3 }}
          {{ $defaultMinReplicas := 2 }}
          {{ $absoluteMaxReplicas := 10 }}
          {{ $defaultTargetUtilizationPercent := 80 }}
          workload: {{ index (splitList "." "${context.res.id}") 1 }}

```

Comment (astromechza) [reply resource-definitions/template-driver/horizontalpodautoscaler/hpa-template.yaml:13]: And then use these constants below.

**Review by TobiasBabin — APPROVED** [review]
LGTM - tested and works. Thanks a bunch!

File: resource-definitions/template-driver/horizontalpodautoscaler/hpa-template.yaml (hunk 2 above)
Comment (mathieu-benoit) [reply resource-definitions/template-driver/horizontalpodautoscaler/hpa-template.yaml:17]: Yup, good point, done.

**Comment by chrishumanitec** [comment]
Note, here is a PR for a fixed HPA Virtual driver: https://github.com/humanitec/resource-registry/pull/128

**Comment by mathieu-benoit** [comment]
> Note, here is a PR for a fixed HPA Virtual driver: [humanitec/resource-registry#128](https://github.com/humanitec/resource-registry/pull/128)

Great to see, thanks!

And like discussed, we could also move forward with this PR, which will be still relevant even when this new virtual driver will be ready.

I'm merging this PR with these examples.

**Pull request merged**